### PR TITLE
test(button + link): add helper methods to create AllVariants stories

### PR DESCRIPTION
**Pull request by dierat**
### Summary:
Shortcut ticket: https://app.shortcut.com/czi-edu/story/160018/reorganize-button-link-and-clickablestyle-stories

The `All Variants` and `Large Variants On Dark Background` stories for the `Button` and `Link` components are almost exactly the same with a couple of small differences. To reduce code clutter and make it easier to maintain these stories, I'm adding some utility methods to generate those stories. I'm also re-adding the available but hidden colors to the "flat" variant in these grids. My reasoning is that now that we have the `Recommended Variants` stories, it's okay for the "all variants" stories to have all the possible colors. (I also have a concern around devs potentially thinking those colors aren't available and making them themselves if they show up in designs. Hopefully no one will need those buttons anyway, but, well, you never know.)

### Test Plan:
Verify there are no visual changes to the `Button` and `Link` stories *except* the "flat" variant now has all the available colors in the `AllVariants` and `LargeVariantsOnDarkBackground` stories.

### CODE REVIEW

**Review by anniehu4 — APPROVED** [review]
nice 👍

File: packages/components/.storybook/buttonUtils.tsx [hunk 1]
```diff
@@ -176,3 +177,94 @@ export const getRecommendedVariants = (
...
+  variant: ButtonProps["variant"],
+  headingColor: "white" | "neutral",
+  buttonChildren: "Button" | "Link",
+  size?: ButtonProps["size"],
+) => {
+  const states = tag === "button" ? buttonStates : linkStates;
+  const icon =
+    variant === "plain" ? (
+      <AddRoundedIcon className={styles.iconButton} purpose="decorative" />
+    ) : null;
```
[the remark this answers is not in the document]
Comment (dierat) [reply packages/components/.storybook/buttonUtils.tsx]: Thanks for this! I had changed this in a different PR but this code had been stashed before that. Updated!

File: packages/components/.storybook/buttonUtils.tsx [hunk 2]
```diff
@@ -176,3 +177,94 @@ export const getRecommendedVariants = (
...
+
+  return (
+    <React.Fragment key={variant}>
+      <Heading size="h2" color={headingColor}>
+        {variant}
+        {size && ` - ${size}`}
+      </Heading>
+      <table className={styles.variant}>
```
[the remark this answers is not in the document]
Comment (dierat) [reply packages/components/.storybook/buttonUtils.tsx]: Updated to use utility classes - thanks for the push!

**Comment by dierat** [comment]
Thanks for all the reviews on these! ❤️

**Pull request merged**